account_invoice_tax_note/models/account_invoice.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, exceptions, models, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def _get_account_tax_groups(self):
+        """
+        Get every account.tax.group used on current recordset.
+        Force the company into context if invoices are not in the user company
+        :return: account.tax.group recordset
+        """
+        invoices = self
+        company = invoices.mapped("company_id")
+        if len(company) > 1:
+            # If we have more than 1 company, it's because a user with
+            # full access (ex: admin) see invoices mixed between companies
+            message = _("To print invoices, they must belong to the same "
+                        "company!")
+            raise exceptions.UserError(message)
+        if company.id != self.env.user.company_id.id:
+            # Update the context only if it's necessary
+            invoices = invoices.with_context(force_company=company.id)
```

Comment:
these company checks may be a bit overkill? Why not doing a `ensure_one()`, since this method is always used in the context of a single invoice.